modules/gltf/gltf_document.cpp

```diff
@@ -3910,6 +3864,80 @@ Error GLTFDocument::_serialize_images(Ref<GLTFState> p_state) {
 	return OK;
 }
 
+Dictionary GLTFDocument::_serialize_image(Ref<GLTFState> p_state, Ref<Image> p_image, const String &p_image_format, float p_lossy_quality, Ref<GLTFDocumentExtension> p_image_save_extension) {
+	Dictionary image_dict;
+	if (p_image->is_compressed()) {
+		p_image->decompress();
+		ERR_FAIL_COND_V_MSG(p_image->is_compressed(), image_dict, "glTF: Image was compressed, but could not be decompressed.");
+	}
+
+	if (p_state->filename.to_lower().ends_with("gltf")) {
+		String relative_texture_dir = "textures";
+		String full_texture_dir = p_state->base_path.path_join(relative_texture_dir);
+		Ref<DirAccess> da = DirAccess::open(p_state->base_path);
+		ERR_FAIL_COND_V(da.is_null(), image_dict);
+
+		if (!da->dir_exists(full_texture_dir)) {
+			da->make_dir(full_texture_dir);
+		}
+		String image_file_name = p_image->get_name();
+		if (p_image_save_extension.is_valid()) {
+			image_file_name = image_file_name + p_image_save_extension->get_image_file_extension();
+			Error err = p_image_save_extension->save_image_at_path(p_state, p_image, full_texture_dir.path_join(image_file_name), p_image_format, p_lossy_quality);
+			ERR_FAIL_COND_V_MSG(err != OK, image_dict, "glTF: Failed to save image in '" + p_image_format + "' format as a separate file, error " + itos(err) + ".");
+		} else if (p_image_format == "PNG") {
+			image_file_name = image_file_name + ".png";
+			p_image->save_png(full_texture_dir.path_join(image_file_name));
+		} else if (p_image_format == "JPEG") {
+			image_file_name = image_file_name + ".jpg";
+			p_image->save_jpg(full_texture_dir.path_join(image_file_name), p_lossy_quality);
+		} else {
+			ERR_FAIL_V_MSG(image_dict, "glTF: Unknown image format '" + p_image_format + "'.");
+		}
+		image_dict["uri"] = relative_texture_dir.path_join(image_file_name).uri_encode();
+	} else {
+		GLTFBufferViewIndex bvi;
+
+		Ref<GLTFBufferView> bv;
+		bv.instantiate();
+
+		const GLTFBufferIndex bi = 0;
+		bv->buffer = bi;
+		bv->byte_offset = p_state->buffers[bi].size();
+		ERR_FAIL_INDEX_V(bi, p_state->buffers.size(), image_dict);
```

Comment:
```suggestion
		ERR_FAIL_INDEX_V(bi, p_state->buffers.size(), image_dict);
		bv->byte_offset = p_state->buffers[bi].size();
```
Assuming, or the first line will crash